### PR TITLE
Add missing service declaration

### PR DESCRIPTION
**Pull request by tifabien**
## Subject

This fix the missing MediaAdminController and GalleryAdminController service declarations.

I am targeting this branch, because only applicable on master branch.

### CODE REVIEW

**Review by VincentLanglet — COMMENTED** [review]
@jordisala1991 Does this mean that every custom CRUDController has to be tag as a service_subscriber with a container set ?

@dmaicher Do you use custom CRUDController with Sonata 4? Did you have to do this ?
Or is done automatically for the dev thanks to autowire/autoconfigure ?

If not, can we do this with a compilerPass in order to avoid extra config for every crud controller ? cc @sonata-project/contributors

**Comment by jordisala1991** [comment]
With autoconfigure and autowire this goes away.

File: src/Resources/config/controllers.php [hunk 1]
```diff
@@ -36,5 +38,15 @@
                 new ReferenceConfigurator('sonata.media.pool'),
             ])
             ->tag('container.service_subscriber')
+            ->call('setContainer', [new ReferenceConfigurator(ContainerInterface::class)])
+
+        ->set(MediaAdminController::class)
```
Comment (jordisala1991) [inline src/Resources/config/controllers.php:43]: assign a name to this please like above

**Comment by jordisala1991** [comment]
The build will get fixed in #2000

File: src/Resources/config/controllers.php (hunk 1 above)
Comment (tifabien) [reply src/Resources/config/controllers.php]: Name added. 
Note that I had to add also an alias.

Comment (VincentLanglet) [reply src/Resources/config/controllers.php]: The `sonata.media.controller.gallery` does not have an alias.
Why the alias would be require here @jordisala1991 ?

Comment (jordisala1991) [reply src/Resources/config/controllers.php]: I don't know if it should be needed. Maybe changing the admin definition to not rely on the controller class, but the service id.

Comment (tifabien) [reply src/Resources/config/controllers.php]: I confirm that if I don't put this alias, the `sonata.media.controller.media.admin` service is not used. Seems related to the Route builder which deduct the controller and this controller is then instanciated by the SF Controller resolver without its correct configuration.

**Review by VincentLanglet — DISMISSED** [review]

File: src/Resources/config/controllers.php (hunk 1 above)
Comment (jordisala1991) [reply src/Resources/config/controllers.php]: You need to change both this files:

https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Resources/config/doctrine_mongodb_admin.php
https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Resources/config/doctrine_orm_admin.php

Concretely those lines:

https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Resources/config/doctrine_mongodb_admin.php#L42
https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Resources/config/doctrine_mongodb_admin.php#L68
https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Resources/config/doctrine_orm_admin.php#L42
https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Resources/config/doctrine_orm_admin.php#L69

**Review by jordisala1991 — CHANGES_REQUESTED** [review]
Waiting for changes

**Comment by tifabien** [comment]
> Waiting for changes

Sorry @jordisala1991 but what do you want for change?

**Comment by jordisala1991** [comment]
> > Waiting for changes
> 
> Sorry @jordisala1991 but what do you want for change?

The ones mentioned here: https://github.com/sonata-project/SonataMediaBundle/pull/2001#discussion_r661331333

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**